doc/man/man5/strsock.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH STRSOCK 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.\"
.\"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
.BR "Linux Fast-STREAMS" ,
.BR streams (5),
consists of a base package that provides basic
.I SVR 4.2 STREAMS
capabilities for
.B Linux
and a number of separate add-on packages.
.PP
OpenSS7
.I STREAMS
add-on packages provide additional functionality and protocol support not traditionally found on
UNIX systems, or which have traditionally fallen into the domain of value-add system software, or,
in this case, where functionality was optional or differently handled in a number of
implementations.
.PP
The
.B @PACKAGE_TITLE@
package is one of the several optional add-on packages that provide enhanced network service support
beyond that provided by the base
.I STREAMS
package.  This package provides support for
.IR STREAMS -based
sockets,
not provided by the base
.I STREAMS
package.
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.I STREAMS
modules and drivers, libraries, programs, initialization scripts, and daemons that
implement the classical
.IR "POSIX Sockets Subsystem" .
.PP
The
.B @PACKAGE_TITLE@
package contains header files and a number of modules and drivers and the
associated documentation originally contained in the
.B Linux Fast-STREAMS
release.
.PP
The package uses the following standard
.I Open Systems Interconnect (OSI)
.[
x200
.]
conforming header files from the
.B strxnet
package:
.IP \- \w'\-\(em'u
.RB < sys/tiuser.h >
Transport Provider Interface (TPI),
.[
TPI Revision 2.0.0
.]
.BR tpi (7)
.PD 0
.IP \-
.RB < sys/tihdr.h >
Transport Provider Interface (TPI),
.[
TPI Revision 2.0.0
.]
.BR tpi (7)
.PD
.PP
The package currently provides the following
.I STREAMS
modules and drivers:
.IP \- \w'\-\(em'u
.B streams_sockmod.ko
.I module
.PD 0
.IP \-
.B streams_socksys.ko
.B /dev/socksys
.PD
.TP
.BR sockmod (4)
The
.BR sockmod (4)
module, when pushed on a TPI Stream, attempts to provide as wide a range of
support for traditional
.I "SVR 4.2"
.[
svr42
.]
socket module and socket system support.
.IP ""
It does so by recognizing input-output controls from three sets:
.BR sockmod (4)
input-output controls;
.BR socksys (4)
input-output controls, and even
.BR socke t(7)
system calls.
.IP ""
This module is incomplete.
.TP
.BR socksys (4)
The
.BR socksys (4)
driver, when opened, provides a TPI Stream that attempts to provide as
wide a range of support for traditional
.I "SVR 4.2"
.[
svr42
.]
socket module and socket system support.
.IP ""
It does so by recognizing input-output controls from three sets:
.BR sockmod (4)
input-output controls;
.BR socksys (4)
input-output controls, and even
.BR socket (7)
system calls.
.IP ""
This module is incomplete.
.PP
The package currently provides the following "STREAMS" libraries:
.IP \- \w'\-\(em'u
.B libsocket.so
.B libsocket.a
.PD 0
.IP \-
.B libsocklib.so
.B libsocklib.a
.PD
.PP
The
.BR libsocket (3)
library implements the following library calls:
.sp
.RS
.TS
lll.
\fBsocket\fP(3)
.TE
.RE
.PP
The
.BR libsocket (3)
library only needs to implement the single library call because file
descriptors that are returned from the
.BR socket (3)
call have been transformed into real sockets (from the viewpoint of the system
call interface) and standard `glibc' calls for the remaining functions are
used.
.PP
The
.BR libsocklib (3)
library implements the following library calls:
.sp
.RS
.TS
lll.
\fBaccept\fP(3)	\fBlisten\fP(3)	\fBsendto\fP(3)
\fBbind\fP(3)	\fBrecv\fP(3)	\fBsetsockopt\fP(3)
\fBconnect\fP(3)	\fBrecvfrom\fP(3)	\fBshutdown\fP(3)
\fBgetpeername\fP(3)	\fBrecvmsg\fP(3)	\fBsocket\fP(3)
\fBgetsockname\fP(3)	\fBsend\fP(3)	\fBsocketpair\fP(3)
\fBgetsockopt\fP(3)	\fBsendmsg\fP(3)
.TE
.RE
.PP
The
.BR libsocklib (3)
library is the older compatiblity library approach to providing sockets for
.I STREAMS
and implements the system calls as library calls for
.I STREAMS
devices, while calling the `glibc' versions for true sockets.
.PP
The
.BR libsockpath (3)
library implements the following library calls:
.sp
.RS
.TS
lll.
\fB\)setsockpath\fP(3)
\fB\)getsockpath\fP(3)
\fB\)endsockpath\fP(3)
\fB\)getsockpathent\fP(3)
\fB\)freesockpathent\fP(3)
.TE
.RE
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I "SVR 4.2"
.[
svr42
.]
.I STREAMS
drivers, modules, libraries, utilities, test programs, daemons, and
development environment for the development and execution of
.B @PACKAGE_TITLE@
trasnport components and applications for the
.I "SVR 4.2"
.I STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package is currently incomplete.
It is being released as a reference point for the community.  If you are interested in completion of
this package, contact
.RB < info@openss7.com >.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package contains the following manual pages:
.\"
.SS "Sockets API Libraries:"
.HP
.BR accept (3)\ \-\ \c
accept a transport connection.
.PD 0
.HP
.BR bind (3)\ \-\ \c
bind a transport endpoint.
.HP
.BR connect (3)\ \-\ \c
establish a transport connection.
.HP
.BR getpeername (3)\ \-\ \c
get name of remote transport endpoint.
.HP
.BR getsockname (3)\ \-\ \c
get name of local transport endpoint.
.HP
.BR getsockopt (3)\ \-\ \c
get a socket option.
.HP
.BR libsocket (3)\ \-\ \c
socket library for use with
.BR socksys (4)
.HP
.BR libsocklib (3)\ \-\ \c
socket library for use with
.BR sockmod (4)
.HP
.BR listen (3)\ \-\ \c
listen for connections on a transport enpoint
.HP
.BR recv (3)\ \-\ \c
receive data on a transport endpoint
.HP
.BR recvfrom (3)\ \-\ \c
receive data from a transport endpoint
.HP
.BR recvmsg (3)\ \-\ \c
receive a message on a transport endpoint
.HP
.BR send (3)\ \-\ \c
send data on a transport endpoint
.HP
.BR sendmsg (3)\ \-\ \c
send a message on a transport endpoint
.HP
.BR sendto (3)\ \-\ \c
send data to a transport endpoint
.HP
.BR setsockopt (3)\ \-\ \c
set a socket option
.HP
.BR shutdown (3)\ \-\ \c
shut down a transport connection
.HP
.BR socket (3)\ \-\ \c
create a socket
.HP
.BR socketpair (3)\ \-\ \c
create a connected pair of sockets
.HP
.BR sockets (3)\ \-\ \c
the
.BR libsocket (3)
library
.HP
.BR socklib (3)\ \-\ \c
the
.BR libsocklib (3)
library
.HP
.BR sockmod (4)\ \-\ \c
socket module
.HP
.BR socksys (4)\ \-\ \c
socket system driver
.PD
.\"
.SS "Socket to Path Library:"
.HP
.BR sockpath (3)\ \-\ \c
the
.BR libsockpath (3)
library
.PD 0
.HP
.BR setsockpath (3)\ \-\ \c
access the
.BR sock2path (5)
database
.HP
.BR getsockpath (3)\ \-\ \c
get next
.BR sock2path (5)
database entry
.HP
.BR endsockpath (3)\ \-\ \c
close
.BR sock2path (5)
database
.HP
.BR getsockpathent (3)\ \-\ \c
get a
.BR sock2path (5)
database entry
.HP
.BR freesockpathent (3)\ \-\ \c
free
.BR sock2path (5)
database entry
.HP
.BR sp_errbuf (3)\ \-\ \c
sockpath error buffer
.HP
.BR sp_error (3)\ \-\ \c
sockpath error
.HP
.BR sp_perror (3)\ \-\ \c
print sockpath error
.HP
.BR sp_sperror (3)\ \-\ \c
sockpath error string
.HP
.BR sock2path (5)\ \-\ \c
sock2path database
.HP
.BR initsock (8)\ \-\ \c
initialize kernel sock2path database
.HP
.BR soconfig (8)\ \-\ \c
manage the kernel sock2path database
.PD
.\"
.SS Miscellaneous
.HP
.BR strsock (5)\ \-\ \c
the @PACKAGE_TITLE@ package
.PD 0
.HP
.BR strsock_mknod (8)\ \-\ \c
create socket system devices
.HP
.BR test-socket (8)\ \-\ \c
test the socket library
.HP
.BR test-socklib (8)\ \-\ \c
test the socklib library
.HP
.BR test-sockmod (8)\ \-\ \c
test the sockmod module
.HP
.BR test-sockpath (8)\ \-\ \c
test the sockpath library
.PD
.\"
.\"
.SH MODULES
.PP
The
.B @PACKAGE@
package provides the following
.I STREAMS
modules:
.HP
.BR sockmod (4)\ \-\ \c
Socket Module.
.\"
.\"
.SH DEVICES
.PP
The
.B @PACKAGE@
package provides the following
.I STREAMS
drivers:
.HP
.BR socksys (4)\ \-\ \c
Socket System driver.
.\"
.\"
.SH PREREQUISITES
.PP
The
.B @PACKAGE@
package request the following packages to be installed:
.TP
.BR streams (5)
The
.B Linux Fast-STREAMS
base package.
.TP
.BR strcompat (5)
The
.B Linux Fast-STREAMS
compatibility modules package.
.TP
.BR strxns (5)
The
.B Linux Fast-STREAMS
X/Open Netowrking Support package.
.TP
.BR strxnet (5)
The
.B Linux Fast-STREAMS
XTI/TLI Library package.
.TP
.BR strnsl (5)
The
.B Linux Fast-STREAMS
Network Selection Library package.
.\"
.\"
.SH NOTICES
.PP
.I Alpha
release are releases that have received little to no testing, or that have been tested
and contains known bugs or defects that make the package unsuitable even for testing.  The purpose
for an
.I alpha
release are the same as for the
.I pre-alpha
release, with the additional purpose that it is an earily release of partially functional code that
has problems that an external developer might be willing to fix themselves and contribute back to
the project.
.PP
.I Alpha
release packages ship containing warnings that executing the code can crash machines and might
possibly do damage to systems upon which it is executed.
.PP
The
.B @PACKAGE_TITLE@
package has unknown defects.  This is an
.I alpha
release.  Some defects could be harmful.  No validation testing whatsoever has been performed by the
.I OpenSS7 Project
on this software.  The software might fail to configure or compile on some systems.  The
.I OpenSS7 Project
recommends that you do not use this software for purposes other than develoment or evaluation, and
then only with great care.  Use at your own risk.  Remember that there is
.BR "NO WARRANTY" .
(See the section
.RI \(lq "NO WARRANTY" \(rq
under the
.RI \(lq "GNU General Public License" \(rq,
in the manual.)
.PP
This software is
.I alpha
software.  As such, it can likely crash your kernel.  Installation of the software can irreparably
mangle your header files or Linux distribution in such a way as to make it unusable.  Crashes could
lock your system and rebooting the system might not repair the problem.  You can possibly lose all
the data on your system.  Because this software can crash your kernel, the resulting unstable system
could destroy computer hardware or peripherals making them unusable.  You could void the warranty on
any system on which you run this software.
.BR "YOU HAVE BEEN WARNED" .
.\"
.\"
.SH BUGS
.PP
The
.B @PACKAGE@
package is compatible with a wide array of production
.B Linux
distributions, kernels, and target architectures, but is considered
.I alpha
grade by
.IR "The OpenSS7 Project" .
At the time of release, the
.B @PACKAGE@
package had no known bugs.  Problems or development requests can be reported to
.RB < bugs@openss7.org >.
Please, however, use the included
.B send-pr
script for generating actionable problem reports or development requests.
Please see the file
.RI \(lq PROBLEMS \(rq
in the release for problem reporting guidelines.
.\"
.\"
.SH "SEE ALSO"
.PP
.BR accept (3),
.BR bind (3),
.BR connect (3),
.BR endsockpath (3),
.BR freesockpathent (3),
.BR getpeername (3),
.BR getsockname (3),
.BR getsockopt (3),
.BR getsockpath (3),
.BR getsockpathent (3),
.BR libsocket (3),
.BR libsocklib (3),
.BR listen (3),
.BR recv (3),
.BR recvfrom (3),
.BR recvmsg (3),
.BR send (3),
.BR sendmsg (3),
.BR sendto (3),
.BR setsockopt (3),
.BR setsockpath (3),
.BR shutdown (3),
.BR socket (3),
.BR socketpair (3),
.BR sockets (3),
.BR socklib (3),
.BR sockpath (3),
.BR sp_errbuf (3),
.BR sp_error (3),
.BR sp_perror (3),
.BR sp_sperror (3),
.BR sockmod (4),
.BR socksys (4),
.BR sock2path (5),
.BR strsock (5),
.BR initsock (8),
.BR soconfig (8),
.BR strsock_mknod (8),
.BR test-socket (8),
.BR test-socklib (8),
.BR test-sockmod (8),
.BR test-sockpath (8).
.\"
.\"
.SH CONFORMANCE
.PP
.IR "UNIX95" ,
.IR "UNIX98" ,
.IR "SUSv2" ,
.[
susv2
.]
.IR "SUSv3" ,
.[
susv3
.]
.IR "SVR 4.2" ,
.[
svr42spg
.]
.IR Solaris \(rg,
.[
solspg
.]
.IR UnixWare \(rg.
.[
uw7spg
.]
.PP
Much of the behavior of
.I SVR 4
sockets within the
.I POSIX Socket Subsystem
made it into
.I POSIX
and particularly into the
.I UNIX 98
specifications from the
.IR OpenGroup .
Consult individual manual pages for additional conformance information.
.\"
.\"
.SH HISTORY
.PP
The first
.IR STREAMS -based
.I Socket Subsystem
was release with
.IR "SVR 4"
.PP
The
.B @PACKAGE@
package was first publicly released (version
.BR 0.9.2.1 )
at the end of 2006.
.\"
.\"
.[
$LIST$
.]
.\"
.TI


NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/util/init/strsock.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strsock:	start and stop strsock modules
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strsock
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		SOCK character devices are present in the /dev directory and \
#		that the STREAMS SOCK kernel modules are configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strsock
# Required-Start: streams strcompat $network
# Required-Stop: streams strcompat $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS Binary Compatibility Modules
# License: GPL
# Description:	This STREAMS SOCK init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS SOCK character
#	devices are present in the /dev directory and that the STREAMS SOCK
#	kernel modules are configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strsock'
config="/etc/default/$name"
desc="the STREAMS SOCK modules"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRSOCK_MKNOD" ; then
    for STRSOCK_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRSOCK_MKNOD ] ; then
	    break
	else
	    STRSOCK_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRSOCK_PRELOAD"       ] || STRSOCK_PRELOAD=""
[ -n "$STRSOCK_DRIVERS"       ] || STRSOCK_DRIVERS="streams-socksys"
[ -n "$STRSOCK_MODULES"       ] || STRSOCK_MODULES="streams-sockmod"
[ -n "$STRSOCK_MAKEDEVICES"   ] || STRSOCK_MAKEDEVICES="yes"
[ -n "$STRSOCK_REMOVEDEVICES" ] || STRSOCK_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRSOCK_MKNOD" ] && STRSOCK_MAKEDEVICES="no"
[ -z "$STRSOCK_MKNOD" ] && STRSOCK_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRSOCK_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRSOCK_MKNOD" -a :"$STRSOCK_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS SOCK devices: "
	$STRSOCK_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRSOCK_MKNOD" -a ":$STRSOCK_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$STRSOCK_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRSOCK_PRELOAD $STRSOCK_DRIVERS $STRSOCK_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.11  2008-10-21 22:42:20  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.10  2008-04-28 22:33:35  brian
# - updated headers for release
#
# Revision 0.9.2.9  2007/08/14 05:17:31  brian
# - GPLv3 header update
#
# Revision 0.9.2.8  2007/03/08 22:42:45  brian
# - correct redirect
#
# Revision 0.9.2.7  2007/03/08 22:12:41  brian
# - update headers
#
# Revision 0.9.2.6  2007/03/08 22:08:10  brian
# - sense of VERBOSE reversed
#
# Revision 0.9.2.5  2007/03/08 20:25:08  brian
# - ubuntu looks for lsb info in init scripts
#
# Revision 0.9.2.4  2006/10/16 00:21:24  brian
# - do not load too many kernel modules on init
#
# Revision 0.9.2.3  2006/10/13 07:00:09  brian
# - load drivers but not modules by default, but remove modules
#
# Revision 0.9.2.2  2006/10/13 04:00:16  brian
# - corrected init scripts and config files
#
# Revision 0.9.2.1  2006/08/23 10:04:12  brian
# - started STREAMS Sockets package
#
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
